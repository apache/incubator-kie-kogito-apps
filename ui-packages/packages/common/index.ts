export {
  default as DataTable
} from './src/components/Organisms/DataTable/DataTable';
export {
  default as KogitoSpinner
} from './src/components/Atoms/KogitoSpinner/KogitoSpinner';
export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as KogitoPageLayout
} from './src/components/Templates/KogitoPageLayout/KogitoPageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as EndpointLink
} from './src/components/Atoms/EndpointLink/EndpointLink';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export * from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './src/components/Atoms/LoadMore/LoadMore';
export {
  default as DomainExplorer
} from './src/components/Organisms/DomainExplorer/DomainExplorer';
export {
  default as DomainExplorerListDomains
} from './src/components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
