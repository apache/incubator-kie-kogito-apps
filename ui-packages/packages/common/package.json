{
  "name": "@kogito-apps/common",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "author": "",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/kiegroup/kogito-apps.git"
  },
  "scripts": {
    "precommit": "lint-staged",
    "build:prod": "yarn run lint && yarn run test:coverage",
    "test:report": "yarn test --ci --reporters=jest-junit",
    "test": "jest --runInBand --ci --reporters=default --reporters=jest-junit",
    "update-snapshot": "jest --updateSnapshot",
    "test:coverage": "yarn test --coverage",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "lint": "tslint -c ./tslint.json --project . './src/**/*.ts{,x}'",
    "format": "prettier --config ../../.prettierrc --check --write './src/**/*.{tsx,ts,js}'",
    "clean": "rimraf dist",
    "codegen": "graphql-codegen"
  },
  "dependencies": {},
  "devDependencies": {
    "@storybook/addon-a11y": "5.3.19",
    "@storybook/addon-actions": "5.3.18",
    "@storybook/addon-backgrounds": "5.3.19",
    "@storybook/addon-console": "1.2.1",
    "@storybook/addon-knobs": "5.3.18",
    "@storybook/addon-links": "5.3.18",
    "@storybook/addon-options": "5.3.18",
    "@storybook/addon-storysource": "5.3.18",
    "@storybook/addon-viewport": "5.3.18",
    "@storybook/react": "5.3.18",
    "@storybook/source-loader": "5.3.18",
    "babel-loader": "8.1.0",
    "react-docgen-typescript-loader": "3.7.2",
    "storybook-react-router": "1.0.8"
  },
  "jest": {
    "preset": "ts-jest/presets/js-with-ts",
    "setupFiles": [
      "../../config/Jest-config/test-shim.js",
      "../../config/Jest-config/test-setup.js",
      "core-js"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "coveragePathIgnorePatterns": [
      "./src/static",
      "./src/graphql"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest",
      "\\.(jpg|jpeg|png|svg)$": "../../config/Jest-config/fileMocks.js"
    },
    "testMatch": [
      "**/tests/*.(ts|tsx|js)"
    ],
    "moduleNameMapper": {
      "\\.(scss|sass|css)$": "identity-obj-proxy"
    }
  },
  "lint-staged": {
    "*.{ts,tsx}": [
      "yarn run format",
      "yarn run lint"
    ]
  }
}
