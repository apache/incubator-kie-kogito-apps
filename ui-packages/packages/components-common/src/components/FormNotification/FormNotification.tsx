import React, { useState } from 'react';
import {
  Alert,
  AlertActionCloseButton,
  AlertActionLink
} from '@patternfly/react-core/dist/js/components/Alert';
import {
  componentOuiaProps,
  OUIAProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';

export interface Notification {
  type: 'success' | 'error';
  message: string;
  details?: string;
  customActions?: Action[];
  close: () => void;
}

export interface Action {
  label: string;
  onClick: () => void;
}

interface IOwnProps {
  notification: Notification;
}

export const FormNotification: React.FC<IOwnProps & OUIAProps> = ({
  notification,
  ouiaId,
  ouiaSafe
}) => {
  const variant = notification.type === 'error' ? 'danger' : 'success';

  const [showDetails, setShowDetails] = useState<boolean>(false);
  return (
    <Alert
      isInline
      title={notification.message}
      variant={variant}
      data-testid={'alert-box'}
      actionLinks={
        <React.Fragment>
          {notification.details && (
            <AlertActionLink
              onClick={() => setShowDetails(!showDetails)}
              data-testid="view-details"
            >
              View details
            </AlertActionLink>
          )}
          {notification.customActions &&
            notification.customActions.length > 0 &&
            notification.customActions.map((customAction, index) => {
              return (
                <AlertActionLink
                  onClick={customAction.onClick}
                  key={index}
                  data-testid="custom-action"
                >
                  {customAction.label}
                </AlertActionLink>
              );
            })}
        </React.Fragment>
      }
      actionClose={
        <AlertActionCloseButton
          onClose={notification.close}
          data-testid={'close-button'}
        />
      }
      {...componentOuiaProps(ouiaId, 'form-notification-alert', ouiaSafe)}
    >
      {showDetails && notification.details && <p>{notification.details}</p>}
    </Alert>
  );
};
