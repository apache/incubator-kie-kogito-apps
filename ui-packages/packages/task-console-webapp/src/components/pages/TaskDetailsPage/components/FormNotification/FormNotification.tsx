import React, { useState } from 'react';
import {
  Alert,
  AlertActionCloseButton,
  AlertActionLink
} from '@patternfly/react-core';
import {
  componentOuiaProps,
  OUIAProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';

export interface Notification {
  type: 'success' | 'error';
  message: string;
  details?: string;
  customAction?: Action;
  close: () => void;
}

export interface Action {
  label: string;
  onClick: () => void;
}

interface IOwnProps {
  notification: Notification;
}

const FormNotification: React.FC<IOwnProps & OUIAProps> = ({
  notification,
  ouiaId,
  ouiaSafe
}) => {
  const variant = notification.type === 'error' ? 'danger' : 'success';

  const [showDetails, setShowDetails] = useState<boolean>(false);

  return (
    <Alert
      isInline
      title={notification.message}
      variant={variant}
      actionLinks={
        <React.Fragment>
          {notification.details && (
            <AlertActionLink onClick={() => setShowDetails(!showDetails)}>
              View details
            </AlertActionLink>
          )}
          {notification.customAction && (
            <AlertActionLink onClick={notification.customAction.onClick}>
              {notification.customAction.label}
            </AlertActionLink>
          )}
        </React.Fragment>
      }
      actionClose={<AlertActionCloseButton onClose={notification.close} />}
      {...componentOuiaProps(ouiaId, 'form-notification-alert', ouiaSafe)}
    >
      {showDetails && notification.details && <p>{notification.details}</p>}
    </Alert>
  );
};

export default FormNotification;
