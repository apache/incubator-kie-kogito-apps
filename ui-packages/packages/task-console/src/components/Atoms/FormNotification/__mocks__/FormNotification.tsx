import React from 'react';

const MockedFormNotification = (): React.ReactElement => {
  return <></>;
};

export default MockedFormNotification;
