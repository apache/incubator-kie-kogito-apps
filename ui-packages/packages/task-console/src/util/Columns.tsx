/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { DataTableColumn } from '@kogito-apps/common';
import TaskDescription from '../components/Atoms/TaskDescription/TaskDescription';
import React from 'react';
import TaskState from '../components/Atoms/TaskState/TaskState';
import Moment from 'react-moment';

export default class Columns {
  static getDefaultColumn = (
    columnPath: string,
    columnLabel: string
  ): DataTableColumn => {
    return {
      path: columnPath,
      label: columnLabel
    };
  };

  static getDateColumn(
    columnPath: string,
    columnLabel: string
  ): DataTableColumn {
    return {
      label: columnLabel,
      path: columnPath,
      bodyCellTransformer: value => (
        <Moment fromNow>{new Date(`${value}`)}</Moment>
      )
    };
  }

  static getTaskDescriptionColumn(): DataTableColumn {
    return {
      label: 'Name',
      path: 'referenceName',
      bodyCellTransformer: (cellValue, rowTask) => (
        <TaskDescription task={rowTask} />
      )
    };
  }

  static getTaskStateColumn(): DataTableColumn {
    return {
      label: 'State',
      path: 'state',
      bodyCellTransformer: (cellValue, rowTask) => <TaskState task={rowTask} />
    };
  }
}
