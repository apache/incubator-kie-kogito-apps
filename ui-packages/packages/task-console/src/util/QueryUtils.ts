/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { IActiveFilters } from '../context/TaskConsoleFilterContext/TaskConsoleFilterContext';
import { User } from '@kogito-apps/common';

const createSearchTextArray = (taskNames: string[]) => {
  const formattedTextArray = [];
  taskNames.forEach(word => {
    formattedTextArray.push({
      referenceName: {
        like: `*${word}*`
      }
    });
  });
  return {
    or: formattedTextArray
  };
};

const createUserAssignmentClause = (currentUser: User) => {
  return {
    or: [
      { actualOwner: { equal: currentUser.id } },
      {
        and: [
          { actualOwner: { isNull: true } },
          {
            not: { excludedUsers: { contains: currentUser.id } }
          },
          {
            or: [
              { potentialUsers: { contains: currentUser.id } },
              { potentialGroups: { containsAny: currentUser.groups } }
            ]
          }
        ]
      }
    ]
  };
};

export const buildTaskInboxWhereArgument = (
  currentUser: User,
  activeFilters: IActiveFilters
) => {
  if (activeFilters.filters) {
    const filtersClause = [];
    if (activeFilters.filters.status.length > 0) {
      filtersClause.push({
        state: { in: activeFilters.filters.status }
      });
    }
    if (activeFilters.filters.taskNames.length > 0) {
      filtersClause.push(
        createSearchTextArray(activeFilters.filters.taskNames)
      );
    }

    if (filtersClause.length > 0) {
      return {
        and: [
          createUserAssignmentClause(currentUser),
          {
            and: filtersClause
          }
        ]
      };
    }
  }
  return createUserAssignmentClause(currentUser);
};
