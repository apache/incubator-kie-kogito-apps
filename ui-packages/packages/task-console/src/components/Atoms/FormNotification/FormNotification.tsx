/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { useState } from 'react';
import {
  Alert,
  AlertActionCloseButton,
  AlertActionLink
} from '@patternfly/react-core';
import { componentOuiaProps, OUIAProps } from '@kogito-apps/common';
import { NotificationType } from '../../../util/Variants';

export interface Notification {
  type: NotificationType;
  message: string;
  details?: string;
  customAction?: Action;
  close: () => void;
}

export interface Action {
  label: string;
  onClick: () => void;
}

interface IOwnProps {
  notification: Notification;
}

const FormNotification: React.FC<IOwnProps & OUIAProps> = ({
  notification,
  ouiaId,
  ouiaSafe
}) => {
  const variant =
    notification.type === NotificationType.ERROR ? 'danger' : 'success';

  const [showDetails, setShowDetails] = useState<boolean>(false);

  return (
    <Alert
      isInline
      title={notification.message}
      variant={variant}
      actionLinks={
        <React.Fragment>
          {notification.details && (
            <AlertActionLink onClick={() => setShowDetails(!showDetails)}>
              View details
            </AlertActionLink>
          )}
          {notification.customAction && (
            <AlertActionLink onClick={notification.customAction.onClick}>
              {notification.customAction.label}
            </AlertActionLink>
          )}
        </React.Fragment>
      }
      actionClose={<AlertActionCloseButton onClose={notification.close} />}
      {...componentOuiaProps(ouiaId, 'form-notification-alert', ouiaSafe)}
    >
      {showDetails && notification.details && <p>{notification.details}</p>}
    </Alert>
  );
};

export default FormNotification;
