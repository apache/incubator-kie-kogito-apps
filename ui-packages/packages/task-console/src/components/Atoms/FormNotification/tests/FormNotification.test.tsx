/**
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import FormNotification, { Notification } from '../FormNotification';
import {
  AlertActionCloseButton,
  AlertActionLink,
  Alert
} from '@patternfly/react-core';
import { getWrapper } from '@kogito-apps/common';
import { NotificationType } from '../../../../util/Variants';

describe('FormNotification test', () => {
  it('Simple notification test', () => {
    const notificationProps: Notification = {
      type: NotificationType.SUCCESS,
      message: 'The form has been submitted',
      close: jest.fn()
    };

    const wrapper = getWrapper(
      <FormNotification notification={notificationProps} />,
      'FormNotification'
    );

    expect(wrapper).toMatchSnapshot();

    const alert = wrapper.find(Alert);

    expect(alert.exists()).toBeTruthy();
    expect(alert.props().variant).toBe('success');

    expect(wrapper.html()).toContain(notificationProps.message);

    const button = wrapper.find(AlertActionCloseButton).find('button');

    button.simulate('click');

    expect(notificationProps.close).toBeCalled();
  });

  it('Notification with details test', async () => {
    const notificationProps: Notification = {
      type: NotificationType.ERROR,
      message: 'The form has been submitted',
      close: jest.fn(),
      details: 'The details here!'
    };

    let wrapper = getWrapper(
      <FormNotification notification={notificationProps} />,
      'FormNotification'
    );
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.html()).toContain(notificationProps.message);

    const alert = wrapper.find(Alert);

    expect(alert.exists()).toBeTruthy();
    expect(alert.props().variant).toBe('danger');

    const button = wrapper.find(AlertActionLink).find('button');

    expect(button.exists()).toBeTruthy();
    expect(button.getDOMNode().innerHTML).toBe('View details');

    button.simulate('click');

    expect(wrapper).toMatchSnapshot();

    wrapper = wrapper.update().find('FormNotification');

    expect(wrapper.html()).toContain(notificationProps.details);
  });

  it('Notification with custom action test', async () => {
    const notificationProps: Notification = {
      type: NotificationType.SUCCESS,
      message: 'The form has been submitted',
      close: jest.fn(),
      customAction: {
        label: 'Custom action',
        onClick: jest.fn()
      }
    };

    const wrapper = getWrapper(
      <FormNotification notification={notificationProps} />,
      'FormNotification'
    );
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.html()).toContain(notificationProps.message);

    const button = wrapper.find(AlertActionLink).find('button');

    expect(button.exists()).toBeTruthy();
    expect(button.getDOMNode().innerHTML).toBe('Custom action');

    button.simulate('click');

    expect(notificationProps.customAction.onClick).toBeCalled();
  });
});
