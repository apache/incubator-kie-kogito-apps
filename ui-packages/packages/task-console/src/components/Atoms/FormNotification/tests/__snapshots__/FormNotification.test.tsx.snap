// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Footer test Snapshot testing 1`] = `
<FormNotification
  closeAction={[MockFunction]}
  message="The form has been submitted"
>
  <Modal
    actions={
      Array [
        <Button
          onClick={[MockFunction]}
          variant="primary"
        >
          OK
        </Button>,
      ]
    }
    appendTo={
      <body
        class="pf-c-backdrop__open"
      >
        <div>
          <div
            class="pf-c-backdrop"
          >
            <div
              class="pf-l-bullseye"
            >
              <div
                aria-describedby="pf-modal-part-2"
                aria-modal="true"
                class="pf-c-modal-box pf-m-sm"
                data-ouia-component-id="0"
                data-ouia-component-type="PF4/ModalContent"
                data-ouia-safe="true"
                id="pf-modal-part-0"
                role="dialog"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      transform=""
                    />
                  </svg>
                </button>
                <header
                  class="pf-c-modal-box__header"
                >
                  <h1
                    class="pf-c-title pf-m-2xl"
                  >
                    Executing Task
                  </h1>
                </header>
                <div
                  class="pf-c-modal-box__body"
                  id="pf-modal-part-2"
                >
                  <div
                    class="pf-c-content"
                  >
                    <p
                      class=""
                      data-pf-content="true"
                    >
                      The form has been submitted
                    </p>
                  </div>
                </div>
                <footer
                  class="pf-c-modal-box__footer"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-primary"
                    data-ouia-component-id="2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
        </div>
      </body>
    }
    aria-describedby=""
    aria-label=""
    aria-labelledby=""
    className=""
    hasNoBodyWrapper={false}
    header={
      <Title
        headingLevel="h1"
        size="2xl"
      >
        Executing Task
      </Title>
    }
    isOpen={true}
    onClose={[MockFunction]}
    ouiaSafe={true}
    showClose={true}
    title=""
    variant="small"
  >
    <Portal
      containerInfo={
        <div>
          <div
            class="pf-c-backdrop"
          >
            <div
              class="pf-l-bullseye"
            >
              <div
                aria-describedby="pf-modal-part-2"
                aria-modal="true"
                class="pf-c-modal-box pf-m-sm"
                data-ouia-component-id="0"
                data-ouia-component-type="PF4/ModalContent"
                data-ouia-safe="true"
                id="pf-modal-part-0"
                role="dialog"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      transform=""
                    />
                  </svg>
                </button>
                <header
                  class="pf-c-modal-box__header"
                >
                  <h1
                    class="pf-c-title pf-m-2xl"
                  >
                    Executing Task
                  </h1>
                </header>
                <div
                  class="pf-c-modal-box__body"
                  id="pf-modal-part-2"
                >
                  <div
                    class="pf-c-content"
                  >
                    <p
                      class=""
                      data-pf-content="true"
                    >
                      The form has been submitted
                    </p>
                  </div>
                </div>
                <footer
                  class="pf-c-modal-box__footer"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-primary"
                    data-ouia-component-id="2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalContent
        actions={
          Array [
            <Button
              onClick={[MockFunction]}
              variant="primary"
            >
              OK
            </Button>,
          ]
        }
        aria-describedby=""
        aria-label=""
        aria-labelledby=""
        boxId="pf-modal-part-0"
        className=""
        descriptorId="pf-modal-part-2"
        hasNoBodyWrapper={false}
        header={
          <Title
            headingLevel="h1"
            size="2xl"
          >
            Executing Task
          </Title>
        }
        isOpen={true}
        labelId="pf-modal-part-1"
        onClose={[MockFunction]}
        ouiaSafe={true}
        showClose={true}
        title=""
        variant="small"
      >
        <Backdrop>
          <div
            className="pf-c-backdrop"
          >
            <FocusTrap
              active={true}
              className="pf-l-bullseye"
              focusTrapOptions={
                Object {
                  "clickOutsideDeactivates": true,
                }
              }
              paused={false}
            >
              <div
                className="pf-l-bullseye"
              >
                <ModalBox
                  aria-describedby="pf-modal-part-2"
                  aria-label=""
                  aria-labelledby=""
                  className=""
                  data-ouia-component-id={0}
                  data-ouia-component-type="PF4/ModalContent"
                  data-ouia-safe={true}
                  id="pf-modal-part-0"
                  style={Object {}}
                  variant="small"
                >
                  <div
                    aria-describedby="pf-modal-part-2"
                    aria-label={null}
                    aria-labelledby={null}
                    aria-modal="true"
                    className="pf-c-modal-box pf-m-sm"
                    data-ouia-component-id={0}
                    data-ouia-component-type="PF4/ModalContent"
                    data-ouia-safe={true}
                    id="pf-modal-part-0"
                    role="dialog"
                    style={Object {}}
                  >
                    <ModalBoxCloseButton
                      onClose={[MockFunction]}
                    >
                      <Button
                        aria-label="Close"
                        className=""
                        onClick={[MockFunction]}
                        variant="plain"
                      >
                        <button
                          aria-disabled={false}
                          aria-label="Close"
                          className="pf-c-button pf-m-plain"
                          data-ouia-component-id={1}
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe={true}
                          disabled={false}
                          onClick={[MockFunction]}
                          type="button"
                        >
                          <TimesIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 352 512"
                              width="1em"
                            >
                              <path
                                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                transform=""
                              />
                            </svg>
                          </TimesIcon>
                        </button>
                      </Button>
                    </ModalBoxCloseButton>
                    <ModalBoxHeader>
                      <header
                        className="pf-c-modal-box__header"
                      >
                        <Title
                          headingLevel="h1"
                          size="2xl"
                        >
                          <h1
                            className="pf-c-title pf-m-2xl"
                          >
                            Executing Task
                          </h1>
                        </Title>
                      </header>
                    </ModalBoxHeader>
                    <ModalBoxBody
                      id="pf-modal-part-2"
                    >
                      <div
                        className="pf-c-modal-box__body"
                        id="pf-modal-part-2"
                      >
                        <TextContent>
                          <div
                            className="pf-c-content"
                          >
                            <Text>
                              <p
                                className=""
                                data-pf-content={true}
                              >
                                The form has been submitted
                              </p>
                            </Text>
                          </div>
                        </TextContent>
                      </div>
                    </ModalBoxBody>
                    <ModalBoxFooter>
                      <footer
                        className="pf-c-modal-box__footer"
                      >
                        <Button
                          key="confirm-selection"
                          onClick={[MockFunction]}
                          variant="primary"
                        >
                          <button
                            aria-disabled={false}
                            aria-label={null}
                            className="pf-c-button pf-m-primary"
                            data-ouia-component-id={2}
                            data-ouia-component-type="PF4/Button"
                            data-ouia-safe={true}
                            disabled={false}
                            onClick={[MockFunction]}
                            type="button"
                          >
                            OK
                          </button>
                        </Button>
                      </footer>
                    </ModalBoxFooter>
                  </div>
                </ModalBox>
              </div>
            </FocusTrap>
          </div>
        </Backdrop>
      </ModalContent>
    </Portal>
  </Modal>
</FormNotification>
`;
